### ISSUE TITLE
search window inside bootstrap modal doesn't work

### ISSUE THREAD

**Issue by spizzi77** [vitmalina/w2ui#1549]
Hy all i notified that using search properties and opening the windows search inside a bootstrap modal is impossible to write inside the input, both IE that firefox.

**Comment by spizzi77** [comment]
Resolved advenced search in w2grid inside modal, inside the toolbar construct of the grid i do this:

```javascript
        toolbar : {items:[
	              { type: 'button',  id: 'item1',  caption: 'Conferma Selezionati', icon: 'glyphicon glyphicon-plus', hint: 'Conferma Recapito' },
	              { type: 'button',  id: 'item2',  caption: 'Elimina Selezionati', icon: 'glyphicon glyphicon-remove', hint: 'Elimina Recapito' },       
	        ],
	        onClick: function (event) {
	        	console.log(event);
	        	if(event.target == 'w2ui-search-advanced'){
	        		$('#w2ui-overlay-searches-'+nameGrid).appendTo($('#idModalBootstrap'));
	        	}
	        },	            
	    }, 
```

Append the window overlay advanced search to the modal bootstrap, it's the same when we have an autocomplete jquery inside a modal....appendTo Modal.
Bye
Andre

**Comment by spizzi77** [comment]
No, this solution isn't ok, or better it's working you can go an advanced search also inside the modal but the advanced window search is appended just under the table and in little screen doesn't work well...
So it isn't ok....

**Comment by mrg2001** [comment]
Can you please create a jsFiddle with the original problem. I will look at the issue.

**Comment by spizzi77** [comment]
Hy Mrg,
Programming i create html inside the php with a class, like as smarty engine template, but with my own, so in general in php i have 
```php
        $html = new Tpl("view/html/esempiW2ui151.html");
        $html -> setTplVar(array(
                "rand" => rand(),
                "server" => $server,
            ));
        $html -> processTpl();
        $html -> printTpl();
        ob_flush(); 
```

So in jsfiddle i can't do it, but i inserted always on my page, my personal project in the toolbar there are buttons "Modal W2Grid 1.4.3 " and "Modal W2Grid 1.5.1" the grids inside are the same grid of the buttons example w2Grid 1.4.3 and example w2Grid 1.5.1 same grids, you can see that both in 1.4.3 that in 1.5.1 the advanced search inside the modal doesn't go instead if you click the Ex.ple w2Grid 1.4.3 or 1.5.1 so in normal window works perfectly.
But this is a know problem i have had also with jquery autocomplete inside the modal, but jquery autocomplete as a method append instead in w2ui i can appen the advanced window but it's not perfect.
Z-index doesn't work....Instead the normal search works perfectly.....
If you want the code of the grid i'll give you but it's the same of my issue for the onClick and onSelect problem in version 1.5.1.
Here's the link where you can see.....The procedure is
click button--->call javascript function that load the php url in the modal content div-->the php create the html and in the html there is also the script tag of the grid.
I will see that you can't write inside the input of the advanced search both in IE that Firefox and Chrome

https://fractalcosmo.com/MvcWebLog/paginaDefault.php

**Comment by spizzi77** [comment]
Hy again Mrg...I also notified that the render : date in the column is giving problem to sort if i insert 

render: 'date:dd/mm/yyyy'

In column dataNascita with sortable:true is giving an error col is undefined...i notified just now, instead 1.4.3 no problem and sort ok...

**Comment by VytautasBoznis** [comment]
Hi,

I had the same problem, and it's not related to w2ui.

It's caused by modal enforced focus in bootstrap. If you're using bootstrap 3 and lover you'll need to modify the bootstrap.js file to comment out the enforced focus (lines 1049 - 1057 and 1005) it's fixed by adding a focus flag to modals in bootstrap 4 alpha+

**Comment by LeicaHabit** [comment]
This message was created automatically by mail delivery software at Junk Email Filter. 

The message from "Vytautas [Masked]" <FWD-737QTYSMENAIYCQANIQDDYCWEIMCQNTAFQQRTYBTEMGNOYBHCWOAFERIFVEGAIZAJRBEQPKM7RUBOX6OFVZUAUYW3EDJYVCBJGZQXMTV4BDUCAUCCVMFU6JUY5V4EV2PIIXB6XRNPIVMIABFEFASAI2QE4MCIWBBHAQFQJJ26JP6AD4ABOAAL2SMMVDIAYECMSQAAIA=@opayq.com> is being returned by server opayq-outbound.junkemailfilter.com on Tue, 20 Jun 2017 03:41:42 -0700

A message that you sent could not be delivered to all of its recipients.

NOTE: And this is important - Junk Email Filter is a front end spam filtering service. We accept email for our customers, filter it, and pass the good email on. Sometimes when we forward email as good the server we forward to rejects it. Generally if you are getting this message it is because the server we forwarded to  * NOT JUNK EMAIL FILTER * has rejected the message.

If you see below the phrase "error from remote mail server" then it is the server who we are forwarding to, NOT US who is rejecting the email. Read the reason carefully and see if you can correct the cause of the rejection.

The following address(es) failed:

  marc@vlitos.net
    host vlitos.net [69.90.161.175]
    SMTP error from remote mail server after RCPT TO:<marc@vlitos.net>:
    554 Security failure

**Comment by spizzi77** [comment]
Hy Vytautas, yes the enforce focus is causing the problem, now i have commented the method and it's working, i must see if there are other complication commented the enforce modal focus..But now if you try in the link under and click "esempi w2ui" in the toolbar and then click "modal w2Grid ADV-SEARCH..." , now it's working...
Bye for now
Andrea 

https://fractalcosmo.com/MvcWebLog/paginaDefault.php

**Comment by spizzi77** [comment]
Hy again Vytautas i gave a look at the question, maybe it's better overwrite the event on the modal show but sincerely i don't know how to capture the the event, btw maybe it's better only comment one line and not all the method inside the bootstrap file....In this way
```javascript

  Modal.prototype.enforceFocus = function () {
    $(document)
      .off('focusin.bs.modal') // guard against infinite focus loop
      .on('focusin.bs.modal', $.proxy(function (e) {
        if (this.$element[0] !== e.target && !this.$element.has(e.target).length) {
          //this.$element.trigger('focus')
        }
      }, this))
  };
```

**Comment by VytautasBoznis** [comment]
Hi, yes it might be better to leave the function intact i just commented it all out becose i didn't know where else it was invoked and did not want any extra focus captures. But as i was going in to the event further i saw that it was mainly used to enforce keyboard controls. So leave it as you see fit.
